### PR TITLE
Dev/oo 283 mail form

### PR DESCRIPTION
**Pull request by mcmpp**

### CODE REVIEW

**Comment by fabianpage** [comment]
Maybe squash and rebase the commits together to have an easier to comprehend history?

File: app/scripts/depots/overview/depotsoverview.controller.js [hunk 1]
```diff
@@ -91,6 +91,18 @@ angular.module('openolitor-admin')
         isDisabled: function() {
           return !$scope.checkboxes.checkedAny;
         }
+      }, {
+        label: gettext('E-Mail Formular'),
+        noEntityText: true,
+        iconClass: 'glyphicon glyphicon-envelope',
+        onExecute: function() {
+          //TODO OO-762 using Mail-Service functionality on Overview
+
```
Comment (fabianpage) [inline app/scripts/depots/overview/depotsoverview.controller.js]: Implementation missing?

File: app/scripts/abotypen/overview/abotypenoverview.controller.js [hunk 2]
```diff
@@ -104,6 +104,18 @@ angular.module('openolitor-admin')
         isDisabled: function() {
           return !$scope.checkboxes.checkedAny;
         }
+      }, {
+        label: gettext('E-Mail Formular'),
+        noEntityText: true,
+        iconClass: 'glyphicon glyphicon-envelope',
+        onExecute: function() {
+          //TODO OO-762 using Mail-Service functionality on Overview
+
```
Comment (fabianpage) [inline app/scripts/abotypen/overview/abotypenoverview.controller.js]: Implementation missing?

File: app/scripts/touren/overview/tourenoverview.controller.js [hunk 3]
```diff
@@ -131,6 +132,18 @@ angular.module('openolitor-admin')
         isDisabled: function() {
           return !$scope.checkboxes.checkedAny;
         }
+      }, {
+        label: gettext('E-Mail Formular'),
+        noEntityText: true,
+        iconClass: 'glyphicon glyphicon-envelope',
+        onExecute: function() {
+          //TODO OO-762 using Mail-Service functionality on Overview
```
Comment (fabianpage) [inline app/scripts/touren/overview/tourenoverview.controller.js]: Implementaion missing?

File: app/scripts/abotypen/overview/abotypenoverview.controller.js [hunk 4]
```diff
@@ -104,6 +127,24 @@ angular.module('openolitor-admin')
         isDisabled: function() {
           return !$scope.checkboxes.checkedAny;
         }
+      }, {
+        label: gettext('E-Mail Formular'),
+        noEntityText: true,
+        iconClass: 'glyphicon glyphicon-envelope',
+        onExecute: function() {
+          $scope.url = 'mailing/sendEmailToAbotypSubscribers';
+          $scope.message = 'Labels allowed to be replaced: \n {{person.anrede}} \n {{person.vorname}} \n {{person.name}} \n {{person.rolle}} \n {{person.kundeId}} \n {{abotyp.name}} \n {{abotyp.beschreibung}}  \n {{abotyp.preis}}  \n {{abotyp.preiseinheit}}  \n {{abotyp.laufzeiteinheit}}';
```
Comment (AlwinEgger) [inline app/scripts/abotypen/overview/abotypenoverview.controller.js]: Translation:
- Text must be in German
- _gettext_ must be injected and be called

File: app/scripts/abotypen/overview/zusatzabotypenoverview.controller.js [hunk 5]
```diff
@@ -103,6 +103,24 @@ angular.module('openolitor-admin')
         isDisabled: function() {
           return !$scope.checkboxes.checkedAny;
         }
+      },{
+        label: gettext('E-Mail Formular'),
+        noEntityText: true,
+        iconClass: 'glyphicon glyphicon-envelope',
+        onExecute: function() {
+          $scope.url = 'mailing/sendEmailToZusatzabotypSubscribers';
+          $scope.message = 'Labels allowed to be replaced: \n {{person.anrede}} \n {{person.vorname}} \n {{person.name}} \n {{person.rolle}} \n {{person.kundeId}} \n {{zusatzabotyp.name}} \n {{zusatzabotyp.beschreibung}}  \n {{zusatzabotyp.preis}}  \n {{zusatzabotyp.preiseinheit}}  \n {{zusatzabotyp.laufzeiteinheit}}';
```
Comment (AlwinEgger) [inline app/scripts/abotypen/overview/zusatzabotypenoverview.controller.js]: Translation...

File: app/scripts/kunden/overview/kundenoverview.controller.js [hunk 6]
```diff
@@ -115,6 +124,25 @@ angular.module('openolitor-admin')
         isDisabled: function() {
           return !$scope.checkboxes.checkedAny;
         }
+      }, {
+        label: gettext('E-Mail Formular'),
+        noEntityText: true,
+        iconClass: 'glyphicon glyphicon-pencil',
+        onExecute: function() {
+          $scope.url = 'mailing/sendEmailToKunden';
+          $scope.message = 'Labels allowed to be replaced: \n {{person.anrede}} \n {{person.vorname}} \n {{person.name}} \n {{person.rolle}} \n {{person.kundeId}} \n {{kunde.bezeichnung}} \n {{kunde.strasse}}  \n {{kunde.hausNummer}}  \n {{kunde.plz}}  \n {{kunde.ort}}';
```
Comment (AlwinEgger) [inline app/scripts/kunden/overview/kundenoverview.controller.js]: Translation...

File: app/scripts/kunden/overview/kundenoverview.controller.js [hunk 7]
```diff
@@ -115,6 +124,25 @@ angular.module('openolitor-admin')
...
+        onExecute: function() {
+          $scope.url = 'mailing/sendEmailToKunden';
+          $scope.message = 'Labels allowed to be replaced: \n {{person.anrede}} \n {{person.vorname}} \n {{person.name}} \n {{person.rolle}} \n {{person.kundeId}} \n {{kunde.bezeichnung}} \n {{kunde.strasse}}  \n {{kunde.hausNummer}}  \n {{kunde.plz}}  \n {{kunde.ort}}';  
+          $scope.kundeIdsMailing = _($scope.filteredEntries)
+            .keyBy('id')
+            .at($scope.checkboxes.ids)
+            .map('id')
+            .value();
+          $scope.showCreateEMailDialog = true;
+          console.log($scope)
```
Comment (AlwinEgger) [inline app/scripts/kunden/overview/kundenoverview.controller.js]: remove

File: app/scripts/touren/overview/tourenoverview.controller.js [hunk 8]
```diff
@@ -131,6 +132,24 @@ angular.module('openolitor-admin')
         isDisabled: function() {
           return !$scope.checkboxes.checkedAny;
         }
+      }, {
+        label: gettext('E-Mail Formular'),
+        noEntityText: true,
+        iconClass: 'glyphicon glyphicon-envelope',
+        onExecute: function() {
+          $scope.url = 'mailing/sendEmailToTourSubscribers';
+          $scope.message = 'Labels allowed to be replaced: \n {{person.anrede}} \n {{person.vorname}} \n {{person.name}} \n {{person.rolle}} \n {{person.kundeId}} \n {{tour.name}} \n {{tour.beschreibung}}';
```
Comment (AlwinEgger) [inline app/scripts/touren/overview/tourenoverview.controller.js]: Translation...

File: bower.json [hunk 9]
```diff
@@ -41,7 +41,7 @@
     "angular-sortable-view": "~0.0.15",
     "angular-css-injector": "^1.4.0",
     "airbrake-js": "^0.5.9",
-    "openolitor-core": "2.2.0-SNAPSHOT",
+    "openolitor-core": "2.2.1-SNAPSHOT",
```
Comment (AlwinEgger) [inline bower.json]: The new core-version

File: package.json [hunk 10]
```diff
@@ -9,7 +9,7 @@
   "license": "GPL-3.0",
   "dependencies": {
     "bower": "1.8.0",
-    "openolitor-core": "git://github.com/OpenOlitor/openolitor-client-core.git#2.2.0-SNAPSHOT",
+    "openolitor-core": "git://github.com/OpenOlitor/openolitor-client-core.git#2.2.1-SNAPSHOT",
```
Comment (AlwinEgger) [inline package.json]: the new core-version

File: app/scripts/kunden/overview/kundenoverview.controller.js (hunk 7 above)
Comment (mcmpp) [reply app/scripts/kunden/overview/kundenoverview.controller.js]: done

File: app/scripts/kunden/overview/kundenoverview.controller.js (hunk 6 above)
Comment (mcmpp) [reply app/scripts/kunden/overview/kundenoverview.controller.js]: done

File: app/scripts/abotypen/overview/zusatzabotypenoverview.controller.js (hunk 5 above)
Comment (mcmpp) [reply app/scripts/abotypen/overview/zusatzabotypenoverview.controller.js]: done

File: app/scripts/abotypen/overview/abotypenoverview.controller.js (hunk 4 above)
Comment (mcmpp) [reply app/scripts/abotypen/overview/abotypenoverview.controller.js]: done

File: package.json (hunk 10 above)
Comment (mcmpp) [reply package.json]: done

File: bower.json (hunk 9 above)
Comment (mcmpp) [reply bower.json]: done

File: app/scripts/touren/overview/tourenoverview.controller.js (hunk 8 above)
Comment (mcmpp) [reply app/scripts/touren/overview/tourenoverview.controller.js]: done

**Review by AlwinEgger — APPROVED** [review]

**Pull request merged**